apps/trading/client-pages/home/home.tsx

```diff
@@ -17,18 +21,37 @@ export const Home = () => {
   const { data, error, loading } = useDataProvider({
     dataProvider: marketsWithDataProvider,
   });
-  const { update } = useGlobalStore((store) => ({
+  const { update, marketId } = useGlobalStore((store) => ({
     update: store.update,
+    marketId: store.marketId,
   }));
-
+  const variables = useMemo(
+    () => ({
+      marketId: marketId || '',
+    }),
+    [marketId]
+  );
+  const { data: singleMarketData, loading: singleMarketLoading } =
+    useDataProvider<SingleMarketFieldsFragment, never>({
+      dataProvider: marketProvider,
+      variables,
+      skip: !marketId,
+    });
+  const marketExists = Boolean(singleMarketData);
+  const marketNotExists = !marketExists && !singleMarketLoading;
   const { pageTitle, updateTitle } = usePageTitleStore((store) => ({
     pageTitle: store.pageTitle,
     updateTitle: store.updateTitle,
   }));
 
   useEffect(() => {
-    if (data) {
-      const marketId = data[0]?.id;
+    if (marketExists) {
+      navigate(Links[Routes.MARKET](marketId as string), {
```

Comment:
chore (blocking): a null check would be better than a type cast (what happens if we call `navigate(null)`)?

Reply:
marketId can't be null here, because marketExists flag is possible to be true only when results of call api with marketId return valid market. I could add there some guard assertion, but such assert type _as string_ seemed simpler to me